### ISSUE TITLE
Pin code structure and UI

### ISSUE THREAD

**Issue by feruzm** [esteemapp/esteem-mobile#13]
Implement Pin code pages and structure.

Light mode: https://zpl.io/Vkn6WXK
Dark mode: https://zpl.io/VKlz6jJ

**AC** (acceptance criteria)

- [x] All below points are only for light mode, theming will be separate issue. But please keep in mind theming while structuring components.
- [x] When app first time launched on any device and user attempts to login, setting pin code will be required and first asked from user to setup.
- [x] After login is successful and user already has set pin code, password/privateKey will be encrypted on device storage with chosen pincode
- [x] Pin code setting should ask for Confirm pin code to be sure before saving pin code on device storage
- [ ] Reset pin code page (access it from Settings page)
- [x] Unlock app page (when application comes from background or opens, user will be prompted to enter pin code when they try to vote/transfer/comment/post/etc. any action that requires use of privateKey/password needs pin code to decrypt password)
- [x] Cache privateKey/password until next background/close of app, so it won't ask pin code every time to decrypt while using app

*PS:* Service/component should be customizable so that we can whitelist some operations that will require pin code only once after app restarts, but some operations that will require pin code every time.
